pkg/compactor/compactor.go

```diff
@@ -34,6 +34,11 @@ import (
 	"github.com/cortexproject/cortex/pkg/util/services"
 )
 
+const (
+	blocksMarkedForDeletionName = "cortex_compactor_blocks_marked_for_deletion_total"
```

Comment:
I'm not sure mixing this metric between compactor and blocks cleaner is a good idea. It it perhaps time for BlocksCleaner to be a separate component from Compactor (not necessarily run separately from compactor, but top-level component in Cortex), as it's doing more and more work. It's unfortunate that blocks cleaner metrics start with "cortex_compactor_" prefix. (But that's a topic for a separate PR.)

Reply:
It could be a separate metric, I don't have a particularly strong opinion. Thoughts @pracucci ?

I guess the question is whether the BlocksCleaner is an implementation detail of the Compactor, or a service in it's own right. I think the reasoning behind putting it in the BlocksCleaner was to avoid having another service to manage.

Reply:
Leaving as-is for now.